src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -239,16 +278,25 @@ int IDebugPendingBreakpoint2.Bind()
 
                                     break;
                                 case enum_BP_LOCATION_TYPE.BPLT_DATA_STRING:
-                                    string address = HostMarshal.GetDataBreakpointStringForIntPtr(_bpRequestInfo.bpLocation.unionmember3);
-                                    if (address.Contains(","))
+                                    string address = HostMarshal.GetDataBreakpointStringForIntPtr(_bpRequestInfo.bpLocation.unionmember3); // this returns "*test" and "&test" (expression, broken) or "0x7ffc7e9513f0" (address, working)
+                                    // test -- need to delete; insert something here
+                                    // need to check for 0x => if no 0x, get the address
+                                    if (address.Contains("0x"))
```

Comment:
>Contains [](http://example.com/codeflow?start=48&length=8)

You need to be much more strict about these checks.

I think what you want is something like:

```C#
// In AD7Property.cs:
static readonly Lazy<Regex> s_propertyDataBreakpointPattern = new Lazy<Regex>(() => new Regex(@"^(0x[0-9a-fA-F]+),.+"));

internal bool TryParsePropertyDataBreakpoint(string input, out string address)
{
     Match match = s_propertyDataBreakpointPattern.Value.Match(input);
     if (!match.Success)
     {
         address = null;
         return false;
     }
     
     address = match.Captures[0].ToString();
     return true;
}
```

```C#
// Top of the class
static readonly Lazy<Regex> s_addressPattern = new Lazy<Regex>(() => new Regex(@"^0x[0-9a-fA-F]+$"));

// Here
string dataBpExpression = HostMarshal.GetDataBreakpointStringForIntPtr(_bpRequestInfo.bpLocation.unionmember3);

if (AD7Property.TryParsePropertyDataBreakpoint(dataBpExpression, out string addressString))
{
    this.AddressId = dataBpExpression;
    _address = addressString;
}
else 
{
    this.AddressId = null;
    dataBpExpression = dataBpExpression.Trim();    

    if (s_addressPattern.Value.Match(dataBpExpression).Success)
    {
       _address = dataBpExpression;
    }
    else
    {
       _address = Address(dataBpExpression);
    }
}
```

Reply:
Ignore this for the second. I think our approach on this is wrong.